autorest/authorization.go

```diff
@@ -258,3 +261,23 @@ func (egta EventGridKeyAuthorizer) WithAuthorization() PrepareDecorator {
 	}
 	return NewAPIKeyAuthorizerWithHeaders(headers).WithAuthorization()
 }
+
+// BasicAuthorizer implements authorization by adding the Authorization HTTP header
+// with the value "Basic <TOKEN>" where <TOKEN> is a base64-encoded personal access token.
+type BasicAuthorizer struct {
```

Comment:
so it seems we are tying this too much with the VSTS requirement. THe Basic Http Scheme is more generic, accepts a base64 encoded username:password token.  https://tools.ietf.org/html/rfc7617

Reply:
so I guess the params here should be username, password ?

Reply:
Hmm well if this is specific to VSTS should we rename the type to `VSTSAuthorizer`, similar to what we have for cognitive services etc?

Reply:
It's not specific to VSTS.
https://tools.ietf.org/html/rfc7617

Reply:
Well according to the RFC the token should be a tuple `userid:password` however in the ctor all we're asking for is a generic `token` parameter which from the VSTS docs looks like a personal access token.  Am I missing something?

Reply:
for VSTS the token passed to the Autorization header is "Basic <any text>:encodBase64(PAT)"
https://docs.microsoft.com/en-us/azure/devops/integrate/get-started/authentication/pats?view=azure-devops

Reply:
I see.  So seems like we should update the ctor to take user-name and password params, and if user-name is empty string we base64 encode `":<password>"` then.  Does everybody agree?

Reply:
@mikaelkrief I've refactored this to take a username/password.  Can you confirm this is working for you?  If so I will merge and tag.

Reply:
@jhendrixMSFT , Yes that working fine for me. You can merge and tag.